apps/emqx/src/emqx_broker.erl

```diff
@@ -518,17 +502,27 @@ handle_call({subscribe, Topic, SubPid, 0}, {From, _Tag}, State) ->
     Result = maybe_add_route(Existed, Topic, From),
     {reply, Result, State};
 handle_call({subscribe, Topic, SubPid, I}, {From, _Tag}, State) ->
-    Existed = ets:member(?SUBSCRIBER, Topic),
-    true = ets:insert(?SUBSCRIBER, [
-        {Topic, {shard, I}},
-        {{shard, Topic, I}, SubPid}
-    ]),
+    %% This will attempt to add a route per each next shard.
+    %% The overhead must be negligible, but the consistency in general
+    %% / race conditions safety is expected to be stronger.
+    Existed = ets:member(?SUBSCRIBER, {shard, Topic, I}),
```

Comment:
Though I'm not really sure that it solves any issues at the moment, I like the idea that each pool worker is only responsible for its subset of shards and doesn't try to check `ets:member(?SUBSCRIBER, Topic)`, since another worker may modify that state in between. 
It completely eliminates any possible race conditions between several workers processing different shards of the same topic. 
The cost is that we add route per each shard (even though it's normally expected to be already added for any shard > 0). 
But this must be a really small overhead..

Reply:
It's conflicting with https://github.com/emqx/emqx/blob/release-55/apps/emqx/src/emqx_router_syncer.erl#L320.
This can happen with this PR, and `ignored` reply will cause badmatch error.
For example:
Shard 0 is busy waiting for deleting a route (e.g. RPC takes much time). Meanwhile, more than max shard size new subscribers are coming.
So, some shard N subscribe call can be definitely processed while there is no route yet and about the same time when other shard 0 subscribe requests are queued for processing/being processed.